packages/core/core/src/BundleGraph.js

```diff
@@ -863,7 +878,61 @@ export default class BundleGraph {
     return size;
   }
 
+  getReferencingBundles(bundle: Bundle): Array<Bundle> {
+    let bundlesSeen: Set<Bundle> = new Set();
```

Comment:
This custom upward dfs can probably be simplified with `graph.traverseAncestors()` or `graph.dfs()` along reference edges